suite/memory/qemu_virt.x

```diff
@@ -4,14 +4,15 @@
 
 MEMORY
 {
-  RAM : ORIGIN = 0x80000000, LENGTH = 64K
+  Flash(rx) : ORIGIN = 0x80000000, LENGTH = 0x100000
+  SRAM(w) : ORIGIN = 0x80100000, LENGTH = 0x20000
 }
 
-REGION_ALIAS("REGION_TEXT", RAM);
-REGION_ALIAS("REGION_RODATA", RAM);
-REGION_ALIAS("REGION_DATA", RAM);
-REGION_ALIAS("REGION_BSS", RAM);
-REGION_ALIAS("REGION_HEAP", RAM);
-REGION_ALIAS("REGION_STACK", RAM);
+REGION_ALIAS("REGION_TEXT", Flash);
+REGION_ALIAS("REGION_RODATA", Flash);
+REGION_ALIAS("REGION_DATA", SRAM);
+REGION_ALIAS("REGION_BSS", SRAM);
+REGION_ALIAS("REGION_HEAP", SRAM);
+REGION_ALIAS("REGION_STACK", SRAM);
```

Comment:
Why do we move to the flash and sram in this case?

Reply:
Because of the inclusion of a json (de)serializer to handle the messages, the suites required size grew larger than the 64K previously available on the qemu machine we used.

So I moved to a bigger memory size for qemu. I occasionally use qemu for quickly testing changes, because it has a much faster execution time. I wanted to avoid the code having enough space to run on qemu but be to large to run on earl grey, therefore I modified this, so that now the linker assumes it can only use the specified space when targeting qemu.

This way if the suite ever grew too large for the earl grey chip it would also throw a linker error when targeting the qemu platform.

But I remove this change since I can see how it may be confusing.

Reply:
You can keep it, I think it makes sense. Maybe just add a comment somewhere explaining this?